nodejs/awsx/ec2/vpc.ts

```diff
@@ -83,59 +78,81 @@ export class Vpc extends pulumi.ComponentResource {
             }, { parent: this, aliases: [{ parent: pulumi.rootStackResource }] });
             this.id = this.vpc.id;
 
-            // Create the appropriate subnets.  Default to a single public and private subnet for each
-            // availability zone if none were specified.
-            const topology = new VpcTopology(name, cidrBlock, numberOfAvailabilityZones);
-            const subnetDescriptions = topology.createSubnets(args.subnets || [
+            const subnets = args.subnets || [
                 { type: "public" },
                 { type: "private" },
-            ]);
-
-            for (let i = 0, n = subnetDescriptions.length; i < n; i++) {
```

Comment:
much of ths logic moved into the VpcTopology class.  basically, and decisions about hte shape of subnets/natgateways/routes moved into the topoloy class.  All that remained here is actually creating the resources.  This allows us to easily produce the topology in unit tests without actually generating the resources.

Reply:
We previously had this for subnets, but not for natgateways.  because our logic for natgateways was dependent on our partitioning algoritym for subnets, this pr had to update it as well.